README: Fix Windows link dependencies

FrostKiwi opened the pull request:
To build under MSYS2 + MinGW64 with `-lcomctl32 -luuid` fails with `undefined reference to `__imp_CoInitializeEx` and similar linking errors.
```
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\msys64\tmp\cc4PrAbe.ltrans0.ltrans.o:C:/msys64/home/Bayonetta2.0/new-horizon/lbs/nativefiledialog-extended/src/nfd_win.cpp:309: undefined reference to `__imp_CoInitializeEx'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\msys64\tmp\cc4PrAbe.ltrans0.ltrans.o:C:/msys64/home/Bayonetta2.0/new-horizon/lbs/nativefiledialog-extended/src/nfd_win.cpp:340: undefined reference to `__imp_CoCreateInstance'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\msys64\tmp\cc4PrAbe.ltrans0.ltrans.o:C:/msys64/home/Bayonetta2.0/new-horizon/lbs/nativefiledialog-extended/src/nfd_win.cpp:330: undefined reference to `__imp_CoTaskMemFree'
collect2.exe: error: ld returned 1 exit status
```
Instead building with `-lole32 -luuid` solves everything. By my logic, the build info for Windows should be changed as well, but I didn't test that and I don't now enough about the Windows SDK to understand why ole32 is needed but not comctl32.
Otherwise, works great <3

btzy commented:
Thanks for the catch!  I'll investigate this issue on my Windows machine and decide if this is the proper fix.

I presume you're trying to build it manually (i.e. without CMake)?  Since the CI automatically does MinGW builds according to the workflow file here: https://github.com/btzy/nativefiledialog-extended/blob/master/.github/workflows/cmake.yml

FrostKiwi commented:
The libnfd.lib library itself I build using Cmake as described in the readme here: https://github.com/btzy/nativefiledialog-extended#standalone-library
I refer to linking libnfd.lib with my final program. This happens "manually" I guess, using the Makefile I wrote. I include the nfd.h file and include the .lib file and additionally -lole32  -luuid.

btzy commented:
I ran the test programs (built with MSVC 19.2), and used [Dependencies](https://github.com/lucasg/Dependencies) to look at the test executables.

![image](https://user-images.githubusercontent.com/6948096/141510115-c91a1b01-f64a-4728-9abb-8e50b95db687.png)

Indeed it links `ole32.dll`.  However, `comctl32.lib` and `uuid.lib` are nowhere to be found, and I do not remember why `uuid.lib` was added in [this commit](https://github.com/btzy/nativefiledialog-extended/commit/d22342a6c34bff7d97e0c515bcccd9b6c346e825).  The `comctl32.lib` was already in the readme when I forked this project from upstream NFD.

I think we should probably remove both `comctl32.lib` and `uuid.lib`, and replace with just `ole32.lib` (and perhaps `shell32.lib` and `kernel32.lib`), both for MSVC and MinGW (so don't create a separate "MinGW" section).

I'm not sure if there were library refactoring for older versions of Windows, but it doesn't seem likely.

(I'm also surprised it links `vcruntime140.lib`, I did not intend to introduce a C++ standard library dependency here.  Ignore this library for now, I'll probably try to get this dependency eliminated.)

FrostKiwi commented:
I don't know what MinGW64 links by default, especially considering I build my Apps with static linking and it seems to work on all PCs without installing any dependencies.
However, `uuid.lib` is needed, at least in my case: 
Without -luuid I get
```
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\msys64\tmp\ccHrXCTW.ltrans0.ltrans.o:<artificial>:(.rdata$.refptr.CLSID_FileOpenDialog[.refptr.CLSID_FileOpenDialog]+0x0): undefined reference to `CLSID_FileOpenDialog'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\msys64\tmp\ccHrXCTW.ltrans0.ltrans.o:<artificial>:(.rdata$.refptr.IID_IFileOpenDialog[.refptr.IID_IFileOpenDialog]+0x0): undefined reference to `IID_IFileOpenDialog'
collect2.exe: error: ld returned 1 exit status
```

btzy commented:
Oh yeah, that makes sense - `uuid` is a static library, which won't show up in the dependencies of the executable.  I think we should change `comctl32.lib` to `ole32.lib` for all Windows then.  Do you mind amending your PR to do that?  (i.e. don't create a separate MinGW section, since MinGW is still Windows.  Just do a single-word change from `comctl32.lib` to `ole32.lib`.)

FrostKiwi commented:
Removed the MinGW category and checked comctl32 to ole32.
I also added "and MinGW64" to the category to show, that we also tested it with that Toolchain. If that's too much or visually noisy, I'll revert to how it was.

btzy commented:
Thanks!  Could we do the following below instead?  (Since "Windows" is the operating system, and both "MSVC" and "MinGW" are compiler toolchains that target the Windows operating system.)  I also see no reason why this code wouldn't work on old non-MinGW64 MinGW either, so let's just call it "MinGW"?

```
### Windows
On Windows (both MSVC and MinGW), ensure ...
```

FrostKiwi commented:
Yeah, I like that version better mine. Updated the PR.

btzy commented:
Thanks for your help!
